src/lib/FreeC/LiftedIR/Converter/Expr.hs

```diff
@@ -0,0 +1,190 @@
+-- | Implements the IR to lifted IR translation for expressions.
+
+module FreeC.LiftedIR.Converter.Expr
+  ( convertExpr
+  )
+where
+
+import           Control.Monad                  ( join )
+import           Data.Maybe                     ( fromMaybe )
+
+import qualified FreeC.IR.Syntax               as IR
+import           FreeC.IR.SrcSpan               ( SrcSpan(NoSrcSpan) )
+import qualified FreeC.LiftedIR.Syntax         as LIR
+import qualified FreeC.LiftedIR.Converter.Type as LIR
+import           FreeC.Environment
+import           FreeC.Environment.LookupOrFail ( lookupAgdaFreshIdentOrFail
+                                                , lookupAgdaValIdentOrFail
+                                                )
+import           FreeC.Environment.Renamer      ( renameAndDefineAgdaVar )
+import           FreeC.Environment.Fresh        ( freshIRQName )
+import           FreeC.Monad.Converter
+import           FreeC.Monad.Reporter           ( reportFatal
+                                                , Message(Message)
+                                                , Severity(Internal)
+                                                )
+
+-- | Converts an expression from IR to lifted IR and lifts it during the
+--   translation.
+convertExpr :: IR.Expr -> Converter LIR.Expr
+convertExpr expr = convertExpr' expr [] []
```

Comment:
For consistency with the functions in `FreeC.LiftedIR.Converter.Type`, the functions in this module should probably also be called `lift*`.